### ISSUE TITLE
Add CRC check for cell voltage readings

### ISSUE THREAD

**Issue by martinjaeger** [LibreSolar/bq769x0_mbed_lib#1]
Currently, CRC is only checked for single readings and not for consecutive reading of values from the register. However, the CRC for voltage readings should also be added.

**Comment by wickeddoc87** [comment]
This is my solution, Buffer size is too large but I don't care at this stage. 
Maybe that's helpful. Thank you for your great work.
```ruby
void BMSupdateVoltages()
{

  int16_t adcVal = 0;
  uint8_t buf[128];
  uint8_t idx = 0;
  boolean bCRCvalid = false;

  int32_t GroessteZellenSpannung = 0;
  int32_t KleinsteZellenSpannung = 5000;

  // read battery pack voltage


  adcVal = (BMSreadRegister(BAT_HI_BYTE) << 8) | BMSreadRegister(BAT_LO_BYTE);
  BMSControls.iBattVoltage = 4.0 * BMSControls.iadcGain * adcVal / 1000.0 + 4 * BMSControls.iadcOffset;

  Wire2.beginTransmission(BMS_I2C_Address);
  Wire2.write(VC1_HI_BYTE);
  Wire2.endTransmission();


  if (BMS_CRC_Enable == 1) {

    if (Wire2.requestFrom(BMS_I2C_Address, (BMSConfig.imaxAnzahlZellen * 4)) == (BMSConfig.imaxAnzahlZellen * 4)) {

      uint16_t crcHi = 0, crcHiSlave = 0, crcLo = 0, crcLoSlave = 0 ;
      while (Wire2.available()) {
        buf[idx] = Wire2.readByte();
        idx++;
      }

      for (uint8_t i = 0; i < BMSConfig.imaxAnzahlZellen; i++)
      {

        crcLoSlave = (buf[(i * 4) + 1]);
        crcHiSlave = (buf[(i * 4) + 3]);

        if (i == 0) {
          crcHi = 0;
          crcLo = 0;
          crcLo = BQCRC8Calc(crcLo, (BMS_I2C_Address << 1) | 1);
          crcLo = BQCRC8Calc(crcLo, buf[(i * 4)]);
          crcHi = BQCRC8Calc(crcHi, buf[(i * 4) + 2]);
        }

        else {
          crcHi = 0;
          crcLo = 0;
          crcLo = BQCRC8Calc(crcLo, buf[(i * 4)]);
          crcHi = BQCRC8Calc(crcHi, buf[(i * 4) + 2]);
        }

        if ((crcHi == crcHiSlave) || (crcLo == crcLoSlave)) {
          bCRCvalid = true;
        }
      }

      if (bCRCvalid == true) {

        for (uint8_t i = 0; i < BMSConfig.imaxAnzahlZellen; i++)     {

          adcVal = (buf[i * 4] & 0b00111111) << 8 | buf[(i * 4) + 2];
          BMSControls.iZellenSpannungen[i] = adcVal * BMSControls.iadcGain / 1000 + BMSControls.iadcOffset;
          if (BMSControls.iZellenSpannungen[i] > GroessteZellenSpannung) {
            GroessteZellenSpannung = BMSControls.iZellenSpannungen[i];
          }
          if (BMSControls.iZellenSpannungen[i] < KleinsteZellenSpannung && BMSControls.iZellenSpannungen[i] > 500) {
            KleinsteZellenSpannung = BMSControls.iZellenSpannungen[i];
          }
        }
        BMSControls.iGroessteZellenSpannung = GroessteZellenSpannung;
        BMSControls.iKleinsteZellenSpannung = KleinsteZellenSpannung;

      }

      else DebugPrintln("CRC Error while Update Voltages");
    }
  }


  else {
    if (Wire2.requestFrom(BMS_I2C_Address, BMSConfig.imaxAnzahlZellen * 2) == BMSConfig.imaxAnzahlZellen * 2) {
      while (Wire2.available()) {
        buf[idx] = Wire2.readByte();
        idx++;
      }
      for (uint8_t i = 0; i < BMSConfig.imaxAnzahlZellen; i++)
      {
        adcVal = (buf[i * 2] & 0b00111111) << 8 | buf[(i * 2) + 1];
        // TODO: check CRC byte buf[1] and buf[3]
        BMSControls.iZellenSpannungen[i] = adcVal * BMSControls.iadcGain / 1000 + BMSControls.iadcOffset;
        if (BMSControls.iZellenSpannungen[i] > GroessteZellenSpannung) {
          GroessteZellenSpannung = BMSControls.iZellenSpannungen[i];
        }
        if (BMSControls.iZellenSpannungen[i] < KleinsteZellenSpannung && BMSControls.iZellenSpannungen[i] > 500) {
          KleinsteZellenSpannung = BMSControls.iZellenSpannungen[i];
        }
      }
      BMSControls.iGroessteZellenSpannung = GroessteZellenSpannung;
      BMSControls.iKleinsteZellenSpannung = KleinsteZellenSpannung;
    }
  }
}

```

**Comment by martinjaeger** [comment]
Solved with commit d301691c.
